src/modules/collection/sagas.ts

```diff
@@ -421,23 +427,171 @@ export function* collectionSaga(legacyBuilderClient: BuilderAPI, client: Builder
 
-      const txHash: string = yield call(sendTransaction, manager, collectionManager =>
-        collectionManager.createCollection(
+      let txHash: string
+
+      if (paymentMethod === PaymentMethod.FIAT) {
+        const wertPublishFeesEnv: string = yield call([config, config.get], 'WERT_PUBLISH_FEES_ENV')
+
+        if (!wertPublishFeesEnv) {
+          throw new Error('Missing WERT_PUBLISH_FEES_ENV')
+        }
+
+        // Wert variables that depend on the desired environment.
+        let partnerId: string
+        let commodity: string
+        let scAddress: string
+        let network: string
+        let origin: string
+
+        switch (wertPublishFeesEnv) {
+          case 'dev':
+            partnerId = '01HRRQQ70YK4SP88GHM9A61P6B'
+            commodity = 'TT'
+            scAddress = '0xe539E0AED3C1971560517D58277f8dd9aC296281'
+            network = 'mumbai'
+            origin = 'https://sandbox.wert.io'
+            break
+          case 'prod':
+            partnerId = '01HR4TB274GD2VNZW0VEAXNHW2'
+            commodity = 'MANA'
+            scAddress = '0x9D32AaC179153A991e832550d9F96441Ea27763A'
+            network = 'polygon'
+            origin = 'https://widget.wert.io'
+            break
+          default:
+            throw new Error('Invalid WERT_PUBLISH_FEES_ENV')
+        }
+
+        // The transaction input data for publishing the collection.
+        const scInputData = new ethers.utils.Interface(manager.abi).encodeFunctionData('createCollection', [
           forwarder.address,
           factory.address,
-          collection.salt!,
+          collection.salt,
           collection.name,
           getCollectionSymbol(collection),
           getCollectionBaseURI(),
           from,
           toInitializeItems(items)
+        ])
+
+        // The amount of MANA to be purchased required to publish the collection is determined by the price of the rarities.
+        // Given that rarities have the same price, we can use the first rarity and multiply it by the amount of items to get the final price.
+        const rarities: ReturnType<typeof getRarities> = yield select(getRarities)
+        const rarity = rarities[0]
+
+        if (!rarity) {
+          throw new Error('Rarity not found')
+        }
+
+        if (!rarity.prices) {
+          throw new Error('Rarity prices not found')
+        }
+
+        const commodityAmount = (() => {
+          const rarityPriceWei = ethers.BigNumber.from(rarity.prices.MANA)
+          const totalPriceWei = rarityPriceWei.mul(items.length)
+          const totalPriceEth = ethers.utils.formatEther(totalPriceWei.toString())
+          const factor = Math.pow(10, 8)
+
+          // Wert supports up to 8 decimal places.
+          // It is important to round up to this amount of decimal places to avoid issues with the widget.
+          return Math.ceil(Number(totalPriceEth) * factor) / factor
+        })()
```

Comment:
It was in fact created for clarity in order to avoid having those many variables polluting the scope of the rest of the function.
I'll move it to a separate function and add a test.